### PR TITLE
ci: add workflow for PR automation

### PR DESCRIPTION
**Pull request by IKatsuba**
Part of #142 

- [x] when a PR is opened, author should be assigned automatically
- [x] when a PR changes a package (e.g. `plugin-eslint`), the corresponding label should be assigned
- [ ] when a PR is linked to an issue, issue status should change to "In review"
- [ ] when a PR is linked to an issue, all of the issue's labels should be added to the PR
- [ ] when a PR is not linked to an issue, it should be added to Project board with "In review" status
- [x] when a PR title is created or changed, it should be checked with Commitlint
- [ ] when a PR cannot be merged because of conflicts, a "has conflicts" label should be added (and removed once conflicts are resolved)

Make three integrations. Another has to be done in another PR.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+🧩 cli: packages/cli/**
+🧩 core: packages/core/**
+🧩 models: packages/models/**
+🧩 nx-plugin: packages/nx-plugin/**
+🧩 eslint-plugin: packages/plugin-eslint/**
+🧩 lighthouse-plugin: packages/plugin-lighthouse/**
+🧩 utils: packages/utils/**
+
+🔬 testing:
+  - packages/**/*.spec.ts
```
Comment (matejchalk) [inline .github/labeler.yml:10]: The test file pattern changed recently. Also, we could have tests in other folders than `packages` as well.

```suggestion
  - **/*.test.ts
```

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - e2e/**
+  - testing-utils/**
+
+🦾 CI/CD: .github/**
```
Comment (matejchalk) [inline .github/labeler.yml:14]: To distuinguish from issue templates, for example:

```suggestion
🦾 CI/CD: .github/workflows/**
```

File: .github/workflows/pr-title-commitlint-check.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    types: [opened, synchronize, edited]
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: npm install
```
Comment (matejchalk) [inline .github/workflows/pr-title-commitlint-check.yml:14]: Clean install is usually a bit faster:

```suggestion
        run: npm ci
```

File: .github/workflows/pr-title-commitlint-check.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      - name: Check commit messages
```
Comment (matejchalk) [inline .github/workflows/pr-title-commitlint-check.yml:15]:
```suggestion
      - name: Check PR title with commitlint
```

**Review by matejchalk — COMMENTED** [review]
Nice work :slightly_smiling_face: Good idea to add the "testing" and "CI/CD" labels, too :+1:

**Review by Tlacenka — COMMENTED** [review]
I created an example PR and tried it out, nice!

The author self-assignment works. ✔️
The PR title commit lint also works well. ✔️
The only issue I see is the recognition of testing/production code/CI paths which were already mentioned by Matěj. 🎨

**Review by BioPhoton — APPROVED** [review]
Nice automations!

**Pull request merged**